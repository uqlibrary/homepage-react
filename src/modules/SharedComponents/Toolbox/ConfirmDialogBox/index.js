export { default as ConfirmationBox } from './components/ConfirmationBox';
