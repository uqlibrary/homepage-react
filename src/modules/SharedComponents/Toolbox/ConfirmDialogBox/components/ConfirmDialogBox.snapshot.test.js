import { ConfirmDialogBox, styles } from './ConfirmDialogBox';

const getProps = (testProps = {}) => ({
    classes: {},
    hideCancelButton: false,
    locale: {
        alternateActionButtonLabel: 'Maybe',
        confirmationTitle: 'Confirmation',
        confirmationMessage: 'Are you sure?',
        cancelButtonLabel: 'No',
        confirmButtonLabel: 'Yes',
    },
    onAction: jest.fn(),
    onCancelAction: jest.fn(),
    onAlternateAction: jest.fn(),
    onRef: jest.fn(),
    showAlternateActionButton: false,
    ...testProps,
});

function setup(testProps = {}) {
    return getElement(ConfirmDialogBox, getProps(testProps));
}

describe('ConfirmDialogBox snapshots tests', () => {
    it('renders component with yes/no buttons', () => {
        const wrapper = setup();
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders component with yes', () => {
        const wrapper = setup({ hideCancelButton: true });
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders component with yes/no/maybe buttons', () => {
        const wrapper = setup({ showAlternateActionButton: true });
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('should have a proper style generator', () => {
        const theme = {
            breakpoints: {
                up: jest.fn(() => 'test1'),
            },
            palette: {
                warning: {
                    main: 'test2',
                    dark: 'test3',
                },
                white: {
                    main: 'test4',
                },
            },
        };
        expect(styles(theme)).toMatchSnapshot();

        delete theme.palette.warning;
        delete theme.palette.white;
        expect(styles(theme)).toMatchSnapshot();

        delete theme.palette;
        expect(styles(theme)).toMatchSnapshot();
    });

    it('renders component with customised locale', () => {
        const wrapper = setup({
            locale: {
                alternateActionButtonLabel: 'ENG: Maybe',
                cancelButtonLabel: 'ENG: No',
                confirmationTitle: 'ENG: Confirmation',
                confirmationMessage: 'ENG: Are you sure?',
                confirmButtonLabel: 'ENG: Yes',
            },
            showAlternateActionButton: true,
        });
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('should call componentWillUnmount and set ref to null', () => {
        const wrapper = setup();
        const componentWillUnmount = jest.spyOn(wrapper.instance(), 'componentWillUnmount');
        wrapper.unmount();
        expect(componentWillUnmount).toHaveBeenCalled();
    });

    it('should show and hide confirmation dialog', () => {
        const wrapper = setup();
        wrapper.instance().showConfirmation();
        expect(wrapper.state().isDialogOpen).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();

        wrapper.instance()._hideConfirmation();
        expect(wrapper.state().isDialogOpen).toBeFalsy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('the ok-equivalent button should work', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            onAction: testFn,
        });
        wrapper
            .find('WithStyles(ForwardRef(Button))')
            .get(0)
            .props.onClick();

        expect(wrapper.state().isDialogOpen).toBeFalsy();
        expect(testFn).toHaveBeenCalled();
    });

    it('the cancel-equivalent button should work', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            onCancelAction: testFn,
        });

        wrapper
            .find('WithStyles(ForwardRef(Button))')
            .get(1)
            .props.onClick();

        expect(wrapper.state().isDialogOpen).toBeFalsy();
        expect(testFn).toHaveBeenCalled();
    });

    it('the alternate action button should work', () => {
        const testFn = jest.fn();
        const wrapper = setup({
            showAlternateActionButton: true,
            onAlternateAction: testFn,
        });

        wrapper
            .find('WithStyles(ForwardRef(Button))')
            .get(1)
            .props.onClick();

        expect(wrapper.state().isDialogOpen).toBeFalsy();
        expect(testFn).toHaveBeenCalled();
    });
});
