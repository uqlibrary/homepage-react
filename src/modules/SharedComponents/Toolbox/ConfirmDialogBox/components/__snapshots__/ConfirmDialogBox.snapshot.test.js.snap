// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialogBox snapshots tests renders component with customised locale 1`] = `
<WithStyles(ForwardRef(Dialog))
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <WithStyles(ForwardRef(DialogTitle))>
    ENG: Confirmation
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      ENG: Are you sure?
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        xsDown={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          autoFocus={true}
          color="primary"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          ENG: Yes
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-testid="alternate-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          ENG: Maybe
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          ENG: No
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes 1`] = `
<WithStyles(ForwardRef(Dialog))
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <WithStyles(ForwardRef(DialogTitle))>
    Confirmation
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Are you sure?
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        xsDown={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          autoFocus={true}
          color="primary"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes/no buttons 1`] = `
<WithStyles(ForwardRef(Dialog))
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <WithStyles(ForwardRef(DialogTitle))>
    Confirmation
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Are you sure?
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        xsDown={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          autoFocus={true}
          color="primary"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`ConfirmDialogBox snapshots tests renders component with yes/no/maybe buttons 1`] = `
<WithStyles(ForwardRef(Dialog))
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <WithStyles(ForwardRef(DialogTitle))>
    Confirmation
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Are you sure?
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        xsDown={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          autoFocus={true}
          color="primary"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          data-testid="alternate-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          Maybe
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`ConfirmDialogBox snapshots tests should have a proper style generator 1`] = `
Object {
  "alternateActionButtonClass": Object {
    "&:hover": Object {
      "backgroundColor": "test3",
    },
    "backgroundColor": "test2",
    "color": "test4",
  },
}
`;

exports[`ConfirmDialogBox snapshots tests should have a proper style generator 2`] = `
Object {
  "alternateActionButtonClass": Object {
    "&:hover": Object {
      "backgroundColor": undefined,
    },
    "backgroundColor": undefined,
    "color": undefined,
  },
}
`;

exports[`ConfirmDialogBox snapshots tests should have a proper style generator 3`] = `
Object {
  "alternateActionButtonClass": Object {
    "&:hover": Object {
      "backgroundColor": undefined,
    },
    "backgroundColor": undefined,
    "color": undefined,
  },
}
`;

exports[`ConfirmDialogBox snapshots tests should show and hide confirmation dialog 1`] = `
<WithStyles(ForwardRef(Dialog))
  open={true}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <WithStyles(ForwardRef(DialogTitle))>
    Confirmation
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Are you sure?
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        xsDown={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          autoFocus={true}
          color="primary"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`ConfirmDialogBox snapshots tests should show and hide confirmation dialog 2`] = `
<WithStyles(ForwardRef(Dialog))
  open={false}
  style={
    Object {
      "padding": 6,
    }
  }
>
  <WithStyles(ForwardRef(DialogTitle))>
    Confirmation
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      Are you sure?
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <Hidden
        xsDown={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        />
      </Hidden>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          autoFocus={true}
          color="primary"
          data-testid="confirm-dialog-box"
          fullWidth={true}
          onClick={[Function]}
        >
          Yes
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm="auto"
        xs={12}
      >
        <WithStyles(ForwardRef(Button))
          color="primary"
          data-testid="cancel-dialog-box"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          No
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
